src/session.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use cookie::Cookie;
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, COOKIE, SET_COOKIE};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Content {
+    #[serde(rename = "__meta__")]
+    meta: HashMap<String, String>,
+    // TODO: Replace String with Cookie
+    // (need to use Serde's serialize_with and deserialize_with)
+    cookies: HashMap<String, String>,
```

Comment:
Might be covered by your TODO, but HTTPie generates an object with `expires`, `path`, `secure` and `value`, not a string. It crashes when trying to load cookies from this format. (It's otherwise willing to load xh's session files, which is great.)